components/Subscribe/index.tsx

```diff
@@ -1,36 +1,73 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
-import { FC, useState } from "react";
+import { FC, useState, useEffect, useContext } from "react";
 import axios from "axios";
+import { Modal } from "../Modal";
+import UserInfoContext from "@/context/UserInfoContext";
+import { validateEmail } from "@/lib/helpers/validateEmail";
+import { SubscribeButton } from "./subscribeButton";
 
-export const Subscribe: FC= () => {
-	const [email, setEmail] = useState<any>("");
+export const Subscribe: FC = () => {
+	const [email, setEmail] = useState<string>("");
 	const [state, setState] = useState<any>("idle");
-	// const [errorMsg, setErrorMsg] = useState<any>(null);
+	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+	const { userInfo } = useContext(UserInfoContext);
+	const isAuth = userInfo?.email;
+	const [errorMsg, setErrorMsg] = useState<string>("");
+
+
+	type SubscribeModal = boolean | null;
+
+	useEffect(() => {
+		if (!localStorage.getItem("subscribeOpen") || localStorage.getItem("subscribeOpen")) {
```

Comment:
What is the logic behind this? If true, or if false, do something? So regardless of outcome do this no matter what?

Reply:
The logic was to check for 3 things: does key "subscribeOpen" exists in localStorage, if it does - is it set to false, and is user authenticated. 
It definitely looks messy. I had a problem with these states, guess I was rushing to get it done. Will fix it